### PR TITLE
Move host to the registry

### PR DESCRIPTION
**Pull request by trisch-me**
## Changes

Refactoring: move host namespace to the registry

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [ ] ~~[CHANGELOG.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CHANGELOG.md) updated for non-trivial changes.~~
* [ ] ~~[schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.~~

### CODE REVIEW

File: docs/resource/host.md [hunk 1]
```diff
@@ -12,45 +12,31 @@ To report host metrics, the `system.*` namespace SHOULD be used.
...
+| [`host.image.id`](../attributes-registry/host.md) | string | VM image ID or host OS image ID. For Cloud, this value is from the provider. | `ami-07b06b442921831e5` | Recommended |
+| [`host.image.name`](../attributes-registry/host.md) | string | Name of the VM image or OS install the host was instantiated from. | `infra-ami-eks-worker-node-7d4ec78312`; `CentOS-8-x86_64-1905` | Recommended |
+| [`host.image.version`](../attributes-registry/host.md) | string | The version string of the VM image or host OS as defined in [Version Attributes](../resource/README.md#version-attributes). | `0.1` | Recommended |
+| [`host.ip`](../attributes-registry/host.md) | string[] | Available IP addresses of the host, excluding loopback interfaces. [1] | `[192.168.1.140, fe80::abc2:4a28:737a:609e]` | Recommended |
+| [`host.mac`](../attributes-registry/host.md) | string[] | Available MAC addresses of the host, excluding loopback interfaces. [2] | `[AC-DE-48-23-45-67, AC-DE-48-23-45-67-01-9F]` | Recommended |
```
Comment (joaopgrassi) [inline docs/resource/host.md:21]: `host-ip` and `host.mac` were Opt-in but now here are Recommended. Probably it was lost when moving to the registry

File: docs/resource/host.md [hunk 2]
```diff
@@ -12,45 +12,31 @@ To report host metrics, the `system.*` namespace SHOULD be used.
...
+| [`host.type`](../attributes-registry/host.md) | string | Type of host. For Cloud, this must be the machine type. | `n1-standard-1` | Recommended |
 
 **[1]:** IPv4 Addresses MUST be specified in dotted-quad notation. IPv6 addresses MUST be specified in the [RFC 5952](https://www.rfc-editor.org/rfc/rfc5952.html) format.
 
 **[2]:** MAC Addresses MUST be represented in [IEEE RA hexadecimal form](https://standards.ieee.org/wp-content/uploads/import/documents/tutorials/eui.pdf): as hyphen-separated octets in uppercase hexadecimal form from most to least significant.
-
-`host.arch` has the following list of well-known values. If one of them applies, then the respective value MUST be used, otherwise a custom value MAY be used.
```
Comment (joaopgrassi) [inline docs/resource/host.md:29]: Please render the table with the `full` syntax so we can have the enum values again here. Check out other markdowns to see an example

File: docs/resource/host.md [hunk 3]
```diff
@@ -12,45 +12,31 @@ To report host metrics, the `system.*` namespace SHOULD be used.
...
 <!-- semconv host.cpu -->
 | Attribute  | Type | Description  | Examples  | Requirement Level |
 |---|---|---|---|---|
-| `host.cpu.cache.l2.size` | int | The amount of level 2 memory cache available to the processor (in Bytes). | `12288000` | Opt-In |
-| `host.cpu.family` | int | Numeric value specifying the family or generation of the CPU. | `6` | Opt-In |
-| `host.cpu.model.id` | int | Model identifier. It provides more granular information about the CPU, distinguishing it from other CPUs within the same family. | `6` | Opt-In |
-| `host.cpu.model.name` | string | Model designation of the processor. | `11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz` | Opt-In |
-| `host.cpu.stepping` | int | Stepping or core revisions. | `1` | Opt-In |
-| `host.cpu.vendor.id` | string | Processor manufacturer identifier. A maximum 12-character string. [1] | `GenuineIntel` | Opt-In |
+| [`host.cache.l2.size`](../attributes-registry/host.md) | int | The amount of level 2 memory cache available to the processor (in Bytes). | `12288000` | Opt-In |
```
Comment (joaopgrassi) [inline docs/resource/host.md:34]: Are these changes intentional? I think not, right?

**Review by joaopgrassi — CHANGES_REQUESTED** [review]

File: model/registry/host.yaml [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        examples: ["192.168.1.140", "fe80::abc2:4a28:737a:609e"]
+      - id: mac
+        type: string[]
+        brief: >
+          Available MAC addresses of the host, excluding loopback interfaces.
+        note: >
+          MAC Addresses MUST be represented in [IEEE RA hexadecimal form](https://standards.ieee.org/wp-content/uploads/import/documents/tutorials/eui.pdf):
+          as hyphen-separated octets in uppercase hexadecimal form from most to least significant.
+        examples: ['AC-DE-48-23-45-67', 'AC-DE-48-23-45-67-01-9F']
+      - id: vendor.id
```
Comment (AlexanderWert) [inline model/registry/host.yaml:89]:
```suggestion
      - id: cpu.vendor.id
```

File: model/registry/host.yaml [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+        type: string
+        brief: >
+          Processor manufacturer identifier. A maximum 12-character string.
+        note: >
+          [CPUID](https://wiki.osdev.org/CPUID) command returns the vendor ID string in EBX, EDX and ECX registers.
+          Writing these to memory in this order results in a 12-character string.
+        examples: [ 'GenuineIntel' ]
+      - id: family
```
Comment (AlexanderWert) [inline model/registry/host.yaml:97]:
```suggestion
      - id: cpu.family
```

File: model/registry/host.yaml [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+        type: int
+        brief: >
+          Numeric value specifying the family or generation of the CPU.
+        examples: [ 6 ]
+      - id: model.id
```
Comment (AlexanderWert) [inline model/registry/host.yaml:102]:
```suggestion
      - id: cpu.model.id
```

File: model/registry/host.yaml [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+        type: int
+        brief: >
+          Model identifier. It provides more granular information about the CPU, distinguishing it from
+          other CPUs within the same family.
+        examples: [ 6 ]
+      - id: model.name
```
Comment (AlexanderWert) [inline model/registry/host.yaml:108]:
```suggestion
      - id: cpu.model.name
```

File: model/registry/host.yaml [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+        type: string
+        brief: >
+          Model designation of the processor.
+        examples: [ '11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz' ]
+      - id: stepping
```
Comment (AlexanderWert) [inline model/registry/host.yaml:113]:
```suggestion
      - id: cpu.stepping
```

File: model/registry/host.yaml [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+        type: int
+        brief: >
+          Stepping or core revisions.
+        examples: [ 1 ]
+      - id: cache.l2.size
```
Comment (AlexanderWert) [inline model/registry/host.yaml:118]:
```suggestion
      - id: cpu.cache.l2.size
```

File: model/resource/host.yaml [hunk 10]
```diff
@@ -5,133 +5,30 @@ groups:
...
-          MAC Addresses MUST be represented in [IEEE RA hexadecimal form](https://standards.ieee.org/wp-content/uploads/import/documents/tutorials/eui.pdf):
-          as hyphen-separated octets in uppercase hexadecimal form from most to least significant.
-        examples: ['AC-DE-48-23-45-67', 'AC-DE-48-23-45-67-01-9F']
+      - ref: host.id
+      - ref: host.type
+      - ref: host.arch
+      - ref: host.image.name
+      - ref: host.image.id
+      - ref: host.image.version
+      - ref: host.ip
```
Comment (AlexanderWert) [inline model/resource/host.yaml:14]:
```suggestion
      - ref: host.ip
         requirement_level: opt_in
```

File: model/resource/host.yaml [hunk 11]
```diff
@@ -5,133 +5,30 @@ groups:
...
+      - ref: host.mac
```
Comment (AlexanderWert) [inline model/resource/host.yaml:15]:
```suggestion
      - ref: host.mac
         requirement_level: opt_in
```

File: docs/resource/host.md (hunk 1 above)
Comment (trisch-me) [reply docs/resource/host.md]: should be fixed now

File: docs/resource/host.md (hunk 2 above)
Comment (trisch-me) [reply docs/resource/host.md]: done

File: docs/resource/host.md (hunk 3 above)
Comment (trisch-me) [reply docs/resource/host.md]: I have moved host.cpu namespace to the host registry as well. So I think this should be correct.

File: model/resource/host.yaml [hunk 12]
```diff
@@ -5,133 +5,32 @@ groups:
...
-        examples: ['ami-07b06b442921831e5']
-      - id: image.version
-        type: string
-        brief: >
-          The version string of the VM image or host OS as defined in
-          [Version Attributes](README.md#version-attributes).
-        examples: ['0.1']
-      - id: ip
-        type: string[]
+      - ref: host.id
```
Comment (AlexanderWert) [inline model/resource/host.yaml:8]: reference to the `host.name` got lost here

**Review by AlexanderWert — CHANGES_REQUESTED** [review]

File: model/resource/host.yaml (hunk 12 above)
Comment (trisch-me) [reply model/resource/host.yaml:8]: thanks, fixed it, also updated with recent changes in host as well

File: model/registry/host.yaml [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+      - id: image.id
+        type: string
+        brief: >
+          VM image ID or host OS image ID. For Cloud, this value is from the provider.
+        examples: ['ami-07b06b442921831e5']
+      - id: image.version
+        type: string
+        brief: >
+          The version string of the VM image or host OS as defined in
+          [Version Attributes](../resource/README.md#version-attributes).
```
Comment (AlexanderWert) [inline model/registry/host.yaml:71]:
```suggestion
          [Version Attributes](/docs/resource/README.md#version-attributes).
```

**Review by AlexanderWert — APPROVED** [review]

**Comment by AlexanderWert** [comment]
@joaopgrassi your comments should be fixed, PTAL

**Review by joaopgrassi — APPROVED** [review]
